nixos/modules/services/web-apps/ackee.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ackee;
+  settings = cfg.defaultSettings // cfg.settings;
+in
+{
+  options.services.ackee = {
+    enable = mkEnableOption "the Ackee analytics tool";
+
+    defaultSettings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
+      default = {
+        ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+        ACKEE_USERNAME = "ackee";
+        ACKEE_PORT = "3000";
+      };
+      example = literalExample "{}";
+      description = ''
+        Settings can be overwritten in <option>settings</option>.
+        To remove all default settings, set <code>defaultSettings = {}</code>.
+      '';
+    };
+
+    settings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
+      default = { };
+      example = literalExample ''
+        {
+          ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+          ACKEE_USERNAME = "ackee";
+          ACKEE_PORT = "3000";
+        }
+      '';
+      description = ''
+        The configuration of Ackee is done through environment variables. The variables defined in
+        this settings will be set on the systemd service.
+
+        The available options can be found in
+        <link xlink:href="https://github.com/electerious/Ackee/blob/v${pkgs.ackee.version}/docs/Options.md">the documentation</link>.
+      '';
+    };
+
+    settingsFile = mkOption {
+      type = types.path;
+      example = "/etc/ackee/credentials";
+      description = ''
+        Environment file as defined in <citerefentry>
+        <refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum>
+        </citerefentry>.
+
+        This file should be owned by root and not be readable by anyone but root.
+        The file complements the <option>settings</option> option and is supposed to contain
+        confidential settings like <literal>ACKEE_PASSWORD</literal>.
+
+        <programlisting>
+          # content of the environment file
+          ACKEE_PASSWORD=verysecretpassword
+        </programlisting>
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Warn on outdated mongodb
+    warnings = mkIf
```

Comment:
You might want to mention to the sysadmin they need to enable `mongodb`, or just do it for them automatically.